src/entities/trades/curve-trade/contracts.ts

```diff
@@ -0,0 +1,135 @@
+  [ChainId.ARBITRUM_ONE as ChainId]: 'https://arb1.arbitrum.io/rpc'
+}
+
+/**
+ *  Construct a new read-only Provider
+ */
+export const getProvider = (chainId: ChainId) => {
+  const host = RPC_PROVIDER_LIST[chainId]
+  return new JsonRpcProvider(host)
+}
+
+export interface GetBestPoolAndOutputParams {
+  tokenInAddress: string
+  tokenOutAddress: string
+  amountIn: BigNumberish
+  chainId: ChainId
+}
+export type GetExchangeRoutingInfoParams = GetBestPoolAndOutputParams
+
+export interface GetBestPoolAndOutputResult {
+  expectedAmountOut: BigNumber
+  poolAddress: string
+  registryExchangeAddress: string
+}
+
+export interface GetExchangeRoutingInfoResults {
+  routes: string[]
+  indices: BigNumber[]
+  expectedAmountOut: BigNumber
+}
+
+/**
+ * Returns the best pool to route a trade through
+ * @param param
+ * @returns
+ */
+export async function getBestCurvePoolAndOutput({
+  amountIn,
+  tokenInAddress,
+  tokenOutAddress,
+  chainId
+}: GetBestPoolAndOutputParams): Promise<GetBestPoolAndOutputResult | undefined> {
+  if (chainId !== ChainId.MAINNET) {
+    throw new Error('Best Pool Find is only available on Mainnet')
+  }
+
+  const addressProviderContract = new Contract(
+    MAINNET_CONTRACTS.addressProvider,
+    ADDRESS_PROVIDER_ABI,
+    getProvider(chainId)
+  )
+
+  // Curve V2 pools
+  const tricryptoCoins = [
+    TOKENS_MAINNET.usdt.address.toLowerCase(),
+    TOKENS_MAINNET.wbtc.address.toLowerCase(),
+    TOKENS_MAINNET.weth.address.toLowerCase()
+  ]
+
+  if (tricryptoCoins.includes(tokenInAddress.toLowerCase()) && tricryptoCoins.includes(tokenOutAddress.toLowerCase())) {
+    throw new Error("This pair can't be exchanged")
+  }
+
+  const registryExchangeAddress = await addressProviderContract.get_address(2, {
+    gasLimit: 100000 // due to Berlin upgrade. See https://github.com/ethers-io/ethers.js/issues/1474
+  })
+
+  const registryExchangeContract = new Contract(registryExchangeAddress, REGISTRY_EXCHANGE_ABI, getProvider(chainId))
+  const [poolAddress, expectedAmountOut] = await registryExchangeContract.get_best_rate(
+    tokenInAddress,
+    tokenOutAddress,
+    amountIn.toString()
+  )
+
+  if (poolAddress === ZERO_ADDRESS) {
+    return
+  }
+
+  return {
+    poolAddress,
+    expectedAmountOut,
+    registryExchangeAddress
+  }
+}
+
+export async function getExchangeRoutingInfo({
+  amountIn,
+  tokenInAddress,
+  tokenOutAddress
+}: GetExchangeRoutingInfoParams): Promise<GetExchangeRoutingInfoResults | undefined> {
+  // Get router
+  const routerContract = new Contract(MAINNET_CONTRACTS.router, CURVE_ROUTER_ABI, getProvider(ChainId.MAINNET))
+
+  try {
+    const params = [tokenInAddress, tokenOutAddress, amountIn.toString()]
+
+    const res = await routerContract.get_exchange_routing(...params, {
+      from: ZERO_ADDRESS
+    })
+
+    const [routes, indices, expectedAmountOut] = res
+
+    return {
+      expectedAmountOut,
+    indices,
+    routes
+}
+  } catch (e) {
+    // console.log(e)
```

Comment:
Shouldn't delete comments and handle error?

Reply:
> Shouldn't delete comments and handle error?

Contract throws an exception if there's no routing information. Are there better ways to handle this case?

Reply:
That's ok. So in this case I would leave just short comment explaining that contract throws an exception or throw new error here. This commented if-else statement is imo confusing.